latex/epoch.tex

```diff
@@ -310,32 +415,114 @@ \subsection{Rewards Distribution}
 
+
+Finally, the reward calculation is presented in
+Figure~\ref{fig:functions:reward-calc}. The calculation is done pool-by-pool.
+The map $\fun{rewardOnePool}$ calculates the rewards given out to each member
+of a given pool (with leader identified by
+the stake key $\var{poolHK}$). The active stake associated to each
+pool member is passed to $\fun{rewardOnePool}$
+via the $\var{actgr}$ variable, which is a subset of hash keys in this pool
+paired with the stake associated with the keys.
+
+This $\fun{rewardOnePool}$ map outputs a finite map and an $\var{avg}$
+value. The finite map is indexed
+by reward addresses. In particular, the subset of reward addresses corresponding
+to registered keys delegating to the pool with leader $\var{poolHK}$.
+A reward address of key $hk$ is paired with the individual reward value
+produced by the $\fun{indivRew}$ map applied to relevant data passed to
+$\fun{rewardOnePool}$ as well as the results of the $\fun{maxPool}$
+and $\fun{poolRew}$ calculations. The $\var{avg}$ value for the stake pool
+(i.e. the pool's ranking) is also calculated by the $\fun{poolRew}$ function.
+
+The $\fun{reward}$ calculation itself uses the $\fun{rewardOnePool}$
+function to reward each pool's members and calculate the pools' moving averages. In order
+to do this, is given the following parameters:
+
+\begin{itemize}
+\item The number of blocks produced by each key this epoch
+\item The protocol constants
+\item The total coin value of rewards to be distributed this epoch
+\item The full delegation state
+\item The set of outputs in the UTxO on the ledger
+\end{itemize}
+
+The output of the $\fun{reward}$ calculation is a pair. The first component is
+a finite map indicating the new rewards assigned to each reward address.
+The second component is the record of every pool's updated moving average.
+
+Within the $\fun{reward}$ calculation, the active stake correspoinding to each
+(active and delegating to an active pool) stake key, $\var{active}$,
+is calculated by adding up the total active stake for that key currently on
+the ledger. The entries of this finite map are then then grouped
+into subsets by the pool each stake key belongs to using the
+$\fun{groupByPool}$ mapping (to give the $\var{pactive}$ value).
+
+Next, a dataset of $\var{results}$ is generated
+by pairing each pool key with the set of rewards calculated for each of the delegators to
+the pool with this key, as well as the moving average for the pool.
+Both of these are calculated using the $\fun{rewardOnePool}$ function.
+To generate the final output of the $\fun{reward}$ calculation,
+we take the union of the all the subsets of the rewards recods of individual
+keys from the range values of the $\var{results}$ map, and pair it with the
+union of all the pool keys (the domain of the $\var{results}$) and the associated
+moving averages for these (the second component of the range of the
+$\var{results}$ finite map).
+
+\begin{todo}
+  N is sigma times what?
```

Comment:
N, the expected value of leader selections for this pool, is the total active stake of the pool, divided by the total active stake of the system, times the number of slots in the epoch.